src/main/java/io/stargate/sgv2/jsonapi/service/projection/ProjectionLayer.java

```diff
@@ -108,6 +101,77 @@ private static ProjectionLayer buildLayers(
     return root;
   }
 
+  /**
+   * Decodes a path string into its constituent segments based on custom rules.
+   *
+   * <p>The decoding process follows these steps:
+   *
+   * <ol>
+   *   <li>
+   *       <p>Scan the encoded path string character-by-character to build segments.
+   *   <li>
+   *       <p>If the scanned character is {@code '.'}, consider the current segment complete, add it
+   *       to the list, and start a new segment.
+   *   <li>
+   *       <p>If the scanned character is {@code '&'}, check for a following character:
+   *       <ul>
+   *         <li>
+   *             <p>If there is no following character, an error is thrown.
+   *         <li>
+   *             <p>If the following character is either {@code '.'} or {@code '&'}, append that
+   *             following character into current segment.
+   *         <li>
+   *             <p>If it is any other character, an error is thrown (only {@code '&'} and {@code
+   *             '.'} are allowed to be escaped).
+   *       </ul>
+   *   <li>
+   *       <p>If the scanned character is neither {@code '&'} nor {@code '.'}, append it to the
+   *       current segment.
+   *   <li>
+   *       <p>When the end of the expression is reached, consider the last segment complete.
+   * </ol>
+   *
+   * <p>For example, the input {@code "pricing.price&.usd"} results in: {@code ["pricing",
+   * "price.usd"]}.
+   *
+   * @param path the encoded path string.
+   * @return an array of decoded segments.
+   */
+  private static String[] decodePath(String path) {
```

Comment:
Since we will be reusing the logic, we need a separate object to represent decoded path. That could be... `ProjectionPath`? Decoding could be a static factory method in that class.